feature/load testing script

lbb opened the pull request:
Resolves #154

lbb committed:
Inital script commit

lbb committed:
Added errcheck

lbb committed:
Added error printing

lbb committed:
Added ineferr fix

lbb committed:
Added jwt flag

lbb committed:
Added node and cluster count flag

lbb committed:
Changed os.Args to flag.Args for propper parsing

lbb committed:
Fixed to early go termination

lbb committed:
Add datacenter flag

lbb committed:
Fixed send Body Get request

lbb committed:
Added waiting for Running State untiladding nodes

lbb committed:
Revert setting async worker count to 1

lbb committed:
Added max async flag

lbb committed:
Added request logging

scruplelesswizard commented on cmd/test-kubermatic-load/main.go:
```diff
@@ -0,0 +1,229 @@
+
+var (
+	jwtFlag         = flag.String("jwt", "", "The String of the Authorization: header")
+	maxNodesFlag    = flag.Int("nodes", 0, "Spcifies the amount of nodes to create in one cluster (nodes*clusters)")
```
Typo

scruplelesswizard commented on cmd/test-kubermatic-load/main.go:
```diff
@@ -0,0 +1,229 @@
+var (
+	jwtFlag         = flag.String("jwt", "", "The String of the Authorization: header")
+	maxNodesFlag    = flag.Int("nodes", 0, "Spcifies the amount of nodes to create in one cluster (nodes*clusters)")
+	maxClustersFlag = flag.Int("clusters", 0, "Spcifies the amount of clusters to deploy")
```
Typo

scruplelesswizard commented on cmd/test-kubermatic-load/main.go:
```diff
@@ -0,0 +1,229 @@
+	jwtFlag         = flag.String("jwt", "", "The String of the Authorization: header")
+	maxNodesFlag    = flag.Int("nodes", 0, "Spcifies the amount of nodes to create in one cluster (nodes*clusters)")
+	maxClustersFlag = flag.Int("clusters", 0, "Spcifies the amount of clusters to deploy")
+	dcFlag          = flag.String("datacenter", "master", "use this to specify a datacenter")
```
Capitalization

scruplelesswizard commented on cmd/test-kubermatic-load/main.go:
```diff
@@ -0,0 +1,229 @@
+	maxNodesFlag    = flag.Int("nodes", 0, "Spcifies the amount of nodes to create in one cluster (nodes*clusters)")
+	maxClustersFlag = flag.Int("clusters", 0, "Spcifies the amount of clusters to deploy")
+	dcFlag          = flag.String("datacenter", "master", "use this to specify a datacenter")
+	maxAsyncFlag    = flag.Int("max-async", 10, "Spcifies the amount of request running at the same time")
```
Typo and Pluralization

scruplelesswizard commented on cmd/test-kubermatic-load/main.go:
```diff
@@ -0,0 +1,229 @@
+			break
+		}
+		log.Println("Waiting for NS to get created ....")
+		time.Sleep(time.Second * 10)
```
Make this configurable

scruplelesswizard commented on cmd/test-kubermatic-load/main.go:
```diff
@@ -0,0 +1,229 @@
+func main() {
+	flag.Parse()
+	printError := func() {
+		log.Printf("Wrong usage usage.\n\n\t %s [up|purge]\n\n", os.Args[0])
```
Typo

lbb committed:
Add NS request sleep flag

lbb committed:
Fix typo in wrong usage text

lbb committed:
Fix typos in flag descriptions

lbb committed:
Add documentation

lbb committed:
Fix async deletion of nodes

scruplelesswizard commented on cmd/test-kubermatic-load/main.go:
```diff
@@ -17,42 +17,48 @@ import (
-	dcFlag          = flag.String("datacenter", "master", "use this to specify a datacenter")
-	maxAsyncFlag    = flag.Int("max-async", 10, "Spcifies the amount of request running at the same time")
+	jwtFlag             = flag.String("jwt-auth", "", "The String of the authorization header")
+	maxNodesFlag        = flag.Int("node-count", 0, "The amount of nodes to create in one cluster (node-count*cluster-count)")
```
`"The number of nodes to create per cluster"`

scruplelesswizard commented on cmd/test-kubermatic-load/main.go:
```diff
@@ -17,42 +17,48 @@ import (
-	maxAsyncFlag    = flag.Int("max-async", 10, "Spcifies the amount of request running at the same time")
+	jwtFlag             = flag.String("jwt-auth", "", "The String of the authorization header")
+	maxNodesFlag        = flag.Int("node-count", 0, "The amount of nodes to create in one cluster (node-count*cluster-count)")
+	maxClustersFlag     = flag.Int("cluster-count", 0, "The amount of clusters to deploy")
```
`"The number of clusters to create"`

scruplelesswizard commented on cmd/test-kubermatic-load/main.go:
```diff
@@ -17,42 +17,48 @@ import (
+	maxNodesFlag        = flag.Int("node-count", 0, "The amount of nodes to create in one cluster (node-count*cluster-count)")
+	maxClustersFlag     = flag.Int("cluster-count", 0, "The amount of clusters to deploy")
+	dcFlag              = flag.String("datacenter-name", "master", "The master dc")
+	maxAsyncFlag        = flag.Int("max-workers", 10, "The amount of request running at the same time")
```
`"Maximum concurrent requests"`

scruplelesswizard commented on cmd/test-kubermatic-load/main.go:
```diff
@@ -17,42 +17,48 @@ import (
+	maxClustersFlag     = flag.Int("cluster-count", 0, "The amount of clusters to deploy")
+	dcFlag              = flag.String("datacenter-name", "master", "The master dc")
+	maxAsyncFlag        = flag.Int("max-workers", 10, "The amount of request running at the same time")
+	retryNSIntervalFlag = flag.Int64("ns-retry-interval", 10, "The amout of time until a NS alive request is send again")
```
`"Duration in seconds to wait between {I really have no idea what we are waiting between}"`

scruplelesswizard commented on cmd/test-kubermatic-load/main.go:
```diff
@@ -17,42 +17,48 @@ import (
+	retryNSIntervalFlag = flag.Int64("ns-retry-interval", 10, "The amout of time until a NS alive request is send again")
 )
 
+// setAuth sets the jwt token int a requests Authorization header field
```
/int/in/

scruplelesswizard reviewed (changes requested):
Please clarify the param descriptions as required

scruplelesswizard commented:
We are still missing usage instructions. This will not be merged until we have some docs on how to use it.

lbb committed:
Merge branch 'develop' into feature/load-testing-script

lbb committed:
Add doc to load balancer script

lbb committed:
Change flag description

lbb committed:
Add doc instructions

lbb committed:
Added domain Flag

lbb committed:
Ignore unused parameter

lbb committed:
Fmt

realfake committed:
Add missing quote

lbb committed:
Add static sleep

lbb committed:
Executed first test

This test was corrupted by executing different taskt on the same cluster (ci and load test), It willl be exeuted again

lbb committed:
Merge branch 'feature/load-testing-script' of https://github.com/kubermatic/api into feature/load-testing-script

lbb committed:
Add second test

Commit:
Remove test result files from API server repo

Commit:
Remove more results files

Commit:
Clean up verbage and typos in usage instructions

Commit:
Add usage instructions for the datacenter-name flag
